File: Lecture Demos/Week 7/2. Inheritance/Python/Circle.py
from Shape import Shape
import math

class Circle(Shape):
    def __init__(self, radius):
        self.radius = radius
        super().__init__(1)
    
    def area(self):
        return math.pi * self.radius * self.radius 
    
    def noCorners(self):
        return 0

File: Lecture Demos (Spring 2024)/Week 7/2. Inheritance/Java/Circle.java
public class Circle extends Shape{
    private double radius;

    public Circle(double radius) {
        super(1);
        this.radius = radius;
    }

    @Override
    public double area() {
        return Math.PI * radius * radius;
    }

    @Override
    public int noCorners() {
        return 0;
    }
}


















// public class Circle extends Shape{
    
//     public Circle() {
//         super(1);
//     }

//     @Override
//     public String getAreaFormula() {
//         return "PI*r^2";
//     }

//     @Override
//     public String getPerimeterFormula() {
//         return "2*PI*r";
//     }

//     @Override
//     public int noCorners() {
//         return 0;
//     }
// }
